src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -221,6 +221,10 @@ public void stop() throws ConnectException {
         log.error("Failed to run query for table {}: {}", querier.toString(), e);
         resetAndRequeueHead(querier);
         return null;
+      } catch (ConnectException e) {
```

Comment:
I don't think this is safe -- won't it mask other types of errors that have nothing to do with connectivity? And do so indefinitely since we'll never pass those `ConnectExceptions` along? We do want tasks to fail eventually, we just want them to retry for some time if they can.